### ISSUE TITLE
incoherent case: expected and found `list[tuple[str, str]]`

### ISSUE THREAD

**Issue by FabriceSalvaire** [astral-sh/ty#3211]
### Summary

ty seems incoherent for this case:

```python
from prompt_toolkit import PromptSession

session = PromptSession()

message = [
    ('', ''),
]
query = session.prompt(message)

# ty [invalid-argument-type]: Argument to bound method `prompt` is incorrect: Expected
#   str |
#   MagicFormattedText |
#   list[tuple[str, str] |
#        tuple[str, str, (MouseEvent, /) -> object]
#       ] |
#   (() -> Any) |
#   None
# found list[tuple[str, str]]
```

**Comment by AlexWaygood** [comment]
Thanks for the report. Please see https://docs.astral.sh/ty/reference/typing-faq/#invariant-generics for why `list[tuple[str, str]]` is not a subtype of `list[tuple[str, str] | tuple[str, str, (MouseEvent, /) -> object]]` (and therefore why we must emit a diagnostic if you try to pass an object of type `list[tuple[str, str]]` to a parameter expecting that type).

We know our diagnostics aren't very helpful for this kind of thing right now. We plan to improve this soon; see https://github.com/astral-sh/ty/issues/163.
